test/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/SubsidyControllerSpec.scala

```diff
@@ -280,6 +282,296 @@ class SubsidyControllerSpec
+            mockAuthWithEnrolmentAndValidEmail(eori1)
+            mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+            mockGet[SubsidyJourney](eori1)(Right(journeyWithReportedPaymentReturningUser.some))
+            mockUpdate[SubsidyJourney](eori1)(Right(journeyWithReportedPaymentReturningUser))
+          }
+
+          val result = performAction(
+            "report-payment-return" -> "true"
+          )
+
+          status(result) shouldBe SEE_OTHER
+          redirectLocation(result) should contain(routes.SubsidyController.getReportedNoCustomSubsidyPage.url)
+        }
+
+        "the user selects 'false' for reported payment returning user" in {
+          inSequence {
+            mockAuthWithEnrolmentAndValidEmail(eori1)
+            mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+            mockGet[SubsidyJourney](eori1)(Right(journeyWithReportedPaymentReturningUser.some))
+            mockUpdate[SubsidyJourney](eori1)(
+              Right(journeyWithReportedPaymentReturningUser.setReportedPaymentReturningUser(false))
+            )
+          }
+
+          val result = performAction(
+            "report-payment-return" -> "false"
+          )
+
+          status(result) shouldBe SEE_OTHER
+          redirectLocation(result) should contain(routes.NoClaimNotificationController.getNoClaimNotification.url)
+        }
+      }
+    }
+
+    "handling request to get returning user page" must {
+
+      def performAction = controller.getReportedPaymentReturningUserPage(
+        FakeRequest(GET, routes.SubsidyController.getReportedPaymentReturningUserPage.url)
+      )
+
+      "call to get session fails" in {
+        inSequence {
+          mockAuthWithEnrolmentAndValidEmail()
+          mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+          mockGetOrCreate[SubsidyJourney](eori1)(Left(ConnectorError(exception)))
+        }
+        assertThrows[Exception](await(performAction))
+      }
+
+      "display the page" when {
+        "a request is made" in {
+          inSequence {
+            mockAuthWithEnrolmentAndValidEmail()
+            mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+            mockGetOrCreate[SubsidyJourney](eori1)(Right(subsidyJourney))
+          }
+          checkPageIsDisplayed(
+            performAction,
+            messageFromMessageKey("reportPaymentReturnJourney.title"),
+            { doc =>
+              doc.select(".govuk-back-link").attr("href") shouldBe routes.AccountController.getAccountPage.url
+              doc
+                .select("form")
+                .attr("action") shouldBe routes.SubsidyController.postReportedPaymentReturningUserPage.url
+            }
+          )
+        }
+      }
+    }
+
+    "handling get to reported no custom subsidy page" must {
+      def performAction(): Future[Result] = controller.getReportedNoCustomSubsidyPage(FakeRequest())
+
+      "render the page with updated form and correct URLs" in {
```

Comment:
please update the name of the test as there is no form on the page. Also, what do you mean by. "correct urls" maybe describe that in the test name?